homeassistant/components/deako/light.py

```diff
@@ -0,0 +1,121 @@
+"""Binary sensor platform for integration_blueprint."""
+import logging
+from typing import Any
+
+from pydeako.deako import Deako
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: ConfigEntry,
+    add_entities: AddEntitiesCallback,
+) -> None:
+    """Configure the platform."""
+    client: Deako = hass.data[DOMAIN][config.entry_id]
+
+    devices = client.get_devices()
+    if len(devices) == 0:
+        # If deako devices are advertising on mdns, we should be able to get at least one device
+        _LOGGER.warning("No devices found from local integration")
+        await client.disconnect()
+        return
+    lights = [DeakoLightSwitch(client, uuid) for uuid in devices]
+    add_entities(lights)
+
+
+class DeakoLightSwitch(LightEntity):
```

Comment:
New Integration have to use `has_entity_name = True`.
https://developers.home-assistant.io/docs/core/entity#has_entity_name-true-mandatory-for-new-integrations

Reply:
Thanks! Didn't catch that.